src/main/java/mandel/gameoflife/Main.java

```diff
@@ -1,8 +1,34 @@
 package mandel.gameoflife;
 
+import java.io.BufferedReader;
+import java.io.InputStreamReader;
+import java.net.URL;
+
 public class Main {
     public static void main(String[] args) {
-        new GridFrame().setVisible(true);
+        try {
+            String filePath = "https://conwaylife.com/patterns/glider.rle";
+
+            URL url = new URL(filePath);
+            BufferedReader reader = new BufferedReader(new InputStreamReader(url.openStream()));
+            StringBuilder content = new StringBuilder();
+            String line;
+            while ((line = reader.readLine()) != null) {
+                content.append(line).append("\n");
+            }
```

Comment:
What does this do?

Reply:
It continuously reads a line from the input file (the rle link) as long as the line is not null. Then it appends the line to a StringBuilder. At the end of the loop, the StringBuilder should contain all the lines from the file. I just moved this into the loadFromRle method because it seemed more efficient and I adjusted the method and tests accordingly.

Reply:
Why don't you just read that whole file all at once?

Reply:
How can I read the whole file at once while still going through it line by line to get the information for the grid? Or is there a better way to get the information from the file without having to go through it line by line?

Reply:
Ok thank you so much. Is there anything else I should change/fix in my code?

Reply:
In left a number of other comments

Reply:
I made some changes and pushed the new code. It looks like it went through on my end, but should I push it again?